File: 2068-check-whether-two-strings-are-almost-equivalent/2068-check-whether-two-strings-are-almost-equivalent.cpp
int init=[] {ios_base::sync_with_stdio(false);cin.tie(0);return 0;}();
class Solution {
public:
    bool checkAlmostEquivalent(string word1, string word2) {
        unordered_map<char, int> charCount;
        for (char key : word1) charCount[key]++; // Count chars in word1
        // Subtract count for characters in word2
        for (char key : word2) charCount[key]--; 
        // Check if the absolute difference exceeds 3 for any character
        for (auto& pair : charCount) {
            if (abs(pair.second) > 3) {
                return false;
            }
        }

        return true;
    }
};

File: 2068-check-whether-two-strings-are-almost-equivalent/2068-check-whether-two-strings-are-almost-equivalent.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public boolean checkAlmostEquivalent(String word1, String word2) {
        Map<Character, Integer> charCount = new HashMap<>();
        // Count characters in word1
        for (char key : word1.toCharArray()) {
            charCount.put(key, charCount.getOrDefault(key, 0) + 1);
        }
        // Subtract count for characters in word2
        for (char key : word2.toCharArray()) {
            charCount.put(key, charCount.getOrDefault(key, 0) - 1);
        }
        // Check if the absolute difference exceeds 3 for any char
        for (int count : charCount.values()) {
            if (Math.abs(count) > 3) {
                return false;
            }
        }

        return true;
    }
}
